docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,230 @@ sidebar_position: 7
 
-If you haven't been explicitly added to the repository you're contributing to, you will need to create a "fork" you own and upload the changes there before opening a pull request (aka "please, BobaLord, merge my code into the official BobaBoard codebase", see [Git Notes](/docs/engineering/legacy-pages/git-notes) ). Unless you know what you're doing and have forked before running `git clone`, follow these instructions to change your `clone` into `fork`.
+1. **Share your work with the project owners**.
 
-1. Go to the **github website** (you must be logged in) and click on the fork button at the top-right corner of the repository you want.
+> You've been working on your own personal copy of the code so far. Once you're ready, you can ask the people involved in the project to take a look at what you've done so you can get feedback. In technical terms this is called 'creating a pull (or merge) request'
 
-   ![The fork button is located on the top right of the page.](/img/legacy/devenv1.png)
+1. **Get feedback and make adjustments**.
 
-2. Take note of the new git address of your forked repository. It should be something like `https://github.com/YOUR-USERNAME/BOBABOARD-REPO.git`. You can find it on the main page of your new forked repo, under code.
+> The owner of the project (and/or your fellow contributors) may suggest changes or point out bugs in your code. This process is called a 'code review'.
 
-   ![The URL of our new repo is under the "Code" dropdown on your repo's main page.](/img/legacy/devenv2.png)
+1. **Success! Your changes are now included in the project.**
 
-3. Run the following commands from your terminal, in the root folder of the repo that you want to upload.
+> It could take more than one round of code reviews but eventually the owner of the project will accept your changes and add them to the codebase. In technical terms, your changes will be 'merged' into the codebase.
 
-```bash   showLineNumbers
-git config --get remote.origin.url | xargs git remote add upstream
-git remote set-url origin YOUR_NEW_REMOTE_URL_HERE
-git fetch
-```
+1. **Keep your code up to date.**
+
+> You did the thing and now you (and everyone else with a copy of the project on their PC) should get the new and updated version of the code - the one that includes your latest edits! In technical terms, you need to 'pull' the changes.
+
+## Guide
+
+The following is a more in-depth guide to the steps one would follow to contribute a simple change to one of Boba's codebases.
+
+### 1. Fork the repository on GitHub
+
+This will create an exact copy of the entire codebase to your own GitHub account. This means all it's files and the history of changes. You own this copy (in terms of permissions) and it's completely separate from the original. No matter what changes you make, the original codebase (often referred to as `upstream`) will remain unchanged.
```

Comment:
I would repeat the name of the concept rather than saying "this".

So, `**Forking** creates an exact copy...`

I also prefer keeping the present tense, but it's not a hard rule right now.

Same below.